edb/lib/cal.edgeql

```diff
@@ -1920,3 +1922,160 @@ std::max(vals: SET OF array<cal::date_duration>) -> OPTIONAL array<cal::date_dur
+            WHEN NOT "inclower" AND NOT "incupper"
+            THEN daterange("lower", "upper", '()')
+            WHEN "inclower" AND NOT "incupper"
+            THEN daterange("lower", "upper", '[)')
+        END
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_datetime>,
+    step: cal::relative_duration
+) -> set of cal::local_datetime
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT
+            generate_series(
+                (
+                    lower(val) + (
+                        CASE WHEN lower_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                (
+                    upper(val) - (
+                        CASE WHEN upper_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                step::interval
+            )::edgedb.timestamp_t
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_date>
+) -> set of cal::local_date
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT
+            generate_series(
+                (
+                    lower(val) + (
+                        CASE WHEN lower_inc(val)
+                            THEN '0'::interval
+                            ELSE 'P1D'::interval
+                        END
+                    )
+                )::timestamp,
+                (
+                    upper(val) - (
+                        CASE WHEN upper_inc(val)
+                            THEN '0'::interval
+                            ELSE 'P1D'::interval
+                        END
+                    )
+                )::timestamp,
+                'P1D'::interval
+            )::edgedb.date_t
+    $$;
+};
+
+
+CREATE FUNCTION
+std::range_unpack(
+    val: range<cal::local_date>,
+    step: cal::date_duration
+) -> set of cal::local_date
+{
+    SET volatility := 'Immutable';
+    USING SQL $$
+        SELECT
+            generate_series(
+                (
+                    lower(val) + (
+                        CASE WHEN lower_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                (
+                    upper(val) - (
+                        CASE WHEN upper_inc(val)
+                            THEN '0'::interval
+                            ELSE step
+                        END
+                    )
+                )::timestamp,
+                step::interval
+            )::edgedb.date_t
+    $$;
+};
+
+# Need to cast edgedb.date_t to date in order for the @> operator to work.
+CREATE FUNCTION std::contains(
+    haystack: range<cal::local_date>,
```

Comment:
The file layout has to do with sequencing DDL in a way so that we can declare types before their usage in functions, etc. The current layout is what we also do for other `std` functions that apply to `cal` types - we defined the extra ones in the same file as `cal` things.